tensorly/regression/cp_plsr.py

```diff
@@ -0,0 +1,230 @@
+        Y : 2D-array of shape (n_samples, n_predictions)
+            labels associated with each sample
+
+        Returns
+        -------
+        self
+        """
+        ## PREPROCESSING
+        # Check that both tensors are coupled along the first mode
+        if T.shape(X)[0] != T.shape(Y)[0]:
+            raise ValueError(
+                "The first modes of X and Y must be coupled and have the same length."
+            )
+        X, Y = T.copy(X), T.copy(Y)
+
+        # Check the shape of X and Y; convert vector Y to a matrix
+        if T.ndim(X) < 2:
+            raise ValueError("X must be at least a 2-mode tensor.")
+        if (T.ndim(Y) != 1) and (T.ndim(Y) != 2):
+            raise ValueError("Only a matrix (2-mode tensor) Y is allowed.")
+        if T.ndim(Y) == 1:
+            Y = T.reshape(Y, (-1, 1))
+
+        # Mean center the data, record info the object
+        self.X_shape = T.shape(X)
+        self.Y_shape = T.shape(Y)
+        self.X_mean = T.mean(X, axis=0)
+        self.Y_mean = T.mean(Y, axis=0)
+        X -= self.X_mean
+        Y -= self.Y_mean
+
+        self.X_factors = [T.zeros((l, self.n_components)) for l in T.shape(X)]
+        self.Y_factors = [
+            T.tile(Y[:, [0]], self.n_components),
+            T.zeros((T.shape(Y)[1], self.n_components)),
+        ]
+
+        ## FITTING EACH COMPONENT
+        for a in range(self.n_components):
+            oldU = T.ones(T.shape(self.Y_factors[0][:, a])) * T.inf
+            for iter in range(self.n_iter_max):
+                Z = T.einsum("i...,i...->...", X, self.Y_factors[0][:, a])
+                Z_comp = (
+                    tucker(Z, [1] * T.ndim(Z))[1] if Z.ndim >= 2 else [Z / T.norm(Z)]
+                )
+                for ii in range(Z.ndim):
+                    self.X_factors[ii + 1] = T.index_update(self.X_factors[ii + 1], T.index[:, a], Z_comp[ii].flatten())
+
+                self.X_factors[0][:, a] = multi_mode_dot(
+                    X, [ff[:, a] for ff in self.X_factors[1:]], range(1, T.ndim(X))
+                )
+                self.Y_factors[1][:, a] = T.dot(T.transpose(Y), self.X_factors[0][:, a])
+                self.Y_factors[1][:, a] /= T.norm(self.Y_factors[1][:, a])
+                self.Y_factors[0][:, a] = T.dot(Y, self.Y_factors[1][:, a])
+                if T.norm(oldU - self.Y_factors[0][:, a]) < self.tol:
+                    if self.verbose:
+                        print(
+                            "Component {}: converged after {} iterations".format(
+                                a, iter
+                            )
+                        )
+                    break
+                oldU = self.Y_factors[0][:, a].copy()
+
+            X -= CPTensor(
+                (None, [ff[:, a].reshape(-1, 1) for ff in self.X_factors])
+            ).to_tensor()
+            Y -= T.dot(
+                T.dot(
+                    T.dot(self.X_factors[0], T.pinv(self.X_factors[0])),
```

Comment:
Isn't `y * pinv(X)` the same as `lstsq(X, y)`? If so, the latter is more stable and efficient.